graphql/image_resolver.go

```diff
@@ -4,19 +4,22 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/model/distro"
 	"github.com/evergreen-ci/evergreen/rest/model"
 	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/utility"
 )
 
 // Distros is the resolver for the distros field.
-func (r *imageResolver) Distros(ctx context.Context, obj *thirdparty.Image) ([]*model.APIDistro, error) {
+func (r *imageResolver) Distros(ctx context.Context, obj *model.APIImage) ([]*model.APIDistro, error) {
 	if obj == nil {
 		return nil, InternalServerError.Send(ctx, "image undefined when attempting to find corresponding distros")
 	}
```

Comment:
I lowkey think we can remove this if-statement. Esp since you'd have to keep rewriting it in every resolver. The task resolvers never check for this condition, for example